Configure size upload limit

ChandraDevs1 opened the pull request:
feat: Add configurable file upload size via settings.json and CLI

- Added `maximum_file_upload_size` parameter to settings.json
- Implemented priority logic: CLI > settings.json > default (200MB)
- Updated run_app.py to handle upload size configuration
- Removed redundant Streamlit CLI code to avoid conflicts

Resolves #155

@jcharkow Please review the PR and let me know if any changes are needed.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Refactor**
  - Improved readability of the application’s startup process by restructuring parameter calls.

- **New Features**
  - Introduced a configuration option to set the maximum file upload limit to 200MB.
  - Expanded and reorganized user documentation to clarify file upload limits and customization methods in local mode.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ChandraDevs1 commented on run_app.py:
```diff
@@ -1,7 +1,29 @@
+    parser.add_argument('--max-upload', type=int, help='Override upload size (MB)')
+    args, unknown = parser.parse_known_args()
+    
+    # Priority: Command-line > settings.json > default
+    size = args.max_upload or get_upload_limit()
```
If the upload size is ≤ 0, a warning is printed, and the default value (DEFAULT_MAX_SIZE) is used.

Ensures the upload size is always a positive number.

ChandraDevs1 replied:
done.

ChandraDevs1 replied:
thanks coderabbitao!

ChandraDevs1 replied:
Is it is good to set the max_upload_limit of the file what you say i think it restrict the user .

ChandraDevs1 replied:
For Most Users: 500MB limit works out-of-the-box so setting a higher default value i think no a good option because 200MB is ensure both small and large file very frequently . and 2nd think adding documentation i don't think so because it's completely depends on user requirements and 3rd i had already given appropriate error message when they get exceed the limit .

jcharkow commented on run_app.py:
```diff
@@ -1,7 +1,38 @@
-        file="app.py", command_line="streamlit run"
-    )
-    # we will create this function inside our streamlit framework
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--max-upload', type=int, help='Override upload size (MB)')
+    args, unknown = parser.parse_known_args()
```
I am not sure if passing this via argparser is appropriate here, might be too confusing for a user as all the other parameters are set with settings.json except for this one. I would just stick with keeping all the settings in settings.json

jcharkow commented on run_app.py:
```diff
@@ -1,7 +1,38 @@
+        size = DEFAULT_MAX_SIZE
+    
+    # Run Streamlit with the configured upload size
+    subprocess.run([
+        "streamlit", "run", "app.py",
+        "--server.maxUploadSize", str(size)
+    ] + unknown)  # Pass through additional arguments
```
Put this in  cli._main_run_clExplicit

jcharkow commented on run_app.py:
```diff
@@ -1,7 +1,38 @@
+import logging
+
+# Configure logging
+logging.basicConfig(level=logging.WARNING)
```
@t0mdavid-m is there another way that we are logging?

jcharkow commented on run_app.py:
```diff
@@ -1,7 +1,38 @@
+logging.basicConfig(level=logging.WARNING)
+
+# Default upload size (MB) if settings.json is missing/invalid
+DEFAULT_MAX_SIZE = 200
```
Might be better to have this in a new file constants.py so all the constants are in one place and import it from there.

jcharkow commented on settings.json:
```diff
@@ -2,6 +2,7 @@
     "app-name": "OpenMS WebApp Template",
     "github-user": "OpenMS",
     "repository-name": "streamlit-template",
+    "maximum_file_upload_size": 200,
```
change to maximum_file_upload_size_mb as coderabbi suggests

ChandraDevs1 commented on run_app.py:
```diff
@@ -1,7 +1,38 @@
-from streamlit.web import cli
+import argparse
```
Dangerous Use of cli.main():

Calling Streamlit’s internal cli.main() directly (instead of using subprocess.run()) can lead to:

Improper cleanup of child processes.

Conflicts with Streamlit’s internal state management.

Missing Error Handling:

No try/except block for FileNotFoundError or invalid JSON in settings.json. these things

ChandraDevs1 commented:
@jcharkow,
I have addressed all the comments you provided in the pull request. If you have any further feedback or notice any issues, please let me know.

Thank you.

jcharkow commented:
Sorry I was not aware that streamlit had a native way to change the file upload size. This would be the preferred approach so thank you for updating that. 

For our sever, we do want to keep the default at 200Mb so please change the value to that. 

If you could also update the documentation as well so that new users of aware of how to configure this when running locally (e.g. here 
![image](https://github.com/user-attachments/assets/1628c707-fd16-48bc-9af9-d1dd4106849b)

to say something like "In local mode file size upload limit can be adjusted through the streamlit's  .streamlit/config.toml file. with the maxUploadSize  parameter"

I am still confused about the other changes to run_app.py. Are these necessary because streamlit should have its own logging mechanisms?

ChandraDevs1 commented:
> Sorry I was not aware that streamlit had a native way to change the file upload size. This would be the preferred approach so thank you for updating that.
> 
> For our sever, we do want to keep the default at 200Mb so please change the value to that.
> 
> If you could also update the documentation as well so that new users of aware of how to configure this when running locally (e.g. here ![image](https://private-user-images.githubusercontent.com/47336288/419909488-1628c707-fd16-48bc-9af9-d1dd4106849b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEyNzQ2MDUsIm5iZiI6MTc0MTI3NDMwNSwicGF0aCI6Ii80NzMzNjI4OC80MTk5MDk0ODgtMTYyOGM3MDctZmQxNi00OGJjLTlhZjktZDFkZDQxMDY4NDliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA2VDE1MTgyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxZTc1ZTAzODRlYWMyMTk4MjQxZGNlYjAxYmFhMjZjNWFmMDAyZjMyOWQzNDI5NTI2MTBkYWM3ZTM1NTk4N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vAKfIUlEorvRGXVlebpRyXjj0koV1WS8MK1OTyMQvMs)
> 
> to say something like "In local mode file size upload limit can be adjusted through the streamlit's .streamlit/config.toml file. with the maxUploadSize parameter"
> 
> I am still confused about the other changes to run_app.py. Are these necessary because streamlit should have its own logging mechanisms?

@jcharkow 
I understand Streamlit has its own logging, but I added custom logging in run_app.py for better error handling and debugging. If it’s confuse or makes more complex, I can remove it
For the documentation update, I suggest handling it in a separate PR to keep things focused. Let me know your thoughts.

jcharkow commented:
> > Sorry I was not aware that streamlit had a native way to change the file upload size. This would be the preferred approach so thank you for updating that.
> > For our sever, we do want to keep the default at 200Mb so please change the value to that.
> > If you could also update the documentation as well so that new users of aware of how to configure this when running locally (e.g. here ![image](https://private-user-images.githubusercontent.com/47336288/419909488-1628c707-fd16-48bc-9af9-d1dd4106849b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEyNzQ2MDUsIm5iZiI6MTc0MTI3NDMwNSwicGF0aCI6Ii80NzMzNjI4OC80MTk5MDk0ODgtMTYyOGM3MDctZmQxNi00OGJjLTlhZjktZDFkZDQxMDY4NDliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA2VDE1MTgyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxZTc1ZTAzODRlYWMyMTk4MjQxZGNlYjAxYmFhMjZjNWFmMDAyZjMyOWQzNDI5NTI2MTBkYWM3ZTM1NTk4N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vAKfIUlEorvRGXVlebpRyXjj0koV1WS8MK1OTyMQvMs)
> > to say something like "In local mode file size upload limit can be adjusted through the streamlit's .streamlit/config.toml file. with the maxUploadSize parameter"
> > I am still confused about the other changes to run_app.py. Are these necessary because streamlit should have its own logging mechanisms?
> 
> @jcharkow I understand Streamlit has its own logging, but I added custom logging in run_app.py for better error handling and debugging. If it’s confuse or makes more complex, I can remove it

@t0mdavid-m what do you think of the extra logging is it necessary?

t0mdavid-m commented:
I dont think the extra logging is necessary at this point. So I would vote for removing it.

ChandraDevs1 commented:
> I dont think the extra logging is necessary at this point. So I would vote for removing it.

ok I will remove .

ChandraDevs1 commented:
@jcharkow please review the PR

jcharkow reviewed (changes requested):
Looking good, I like the documentation improvements! Please revert the run_app.py to the original form. as the changes are not needed. Then should be ready to merge!

ChandraDevs1 commented on run_app.py:
```diff
@@ -1,7 +1,6 @@
-    )
-    # we will create this function inside our streamlit framework
+    sys.argv = ["streamlit", "run", "app.py"]
+    main()
```
done

ChandraDevs1 reviewed (commented):
all suggestions are implemented please review the PR

ChandraDevs1 commented on docs/user_guide.md:
```diff
@@ -15,15 +15,24 @@ In the OpenMS web application, workspaces are designed to keep your analysis org
-  - *Local Mode*: Multiple file uploads are supported, giving you flexibility when working with large datasets.
-- **Workspace Access**:
-  - In online mode, workspaces are stored temporarily and will be cleared after seven days of inactivity.
-  - In local mode, workspaces are saved on your local machine, allowing for persistent storage.
```
done

jcharkow reviewed (approved):
looks good ready to merge
